portfolio/src/main/webapp/script.js

```diff
@@ -79,29 +79,47 @@ async function fetchComments() {
     const commentsContainer = document.getElementById('comments-container');
     commentsContainer.innerHTML = '';
 
+    console.log(comments);
+
     comments.forEach( comment => {
-        const { username, text, timestamp } = comment;
+        const { username, text, timestamp, imageUrl } = comment;
         commentsContainer.appendChild(
-            createListElement(username, text, timestamp)
+            createListElement(username, text, timestamp, imageUrl)
         );
     });
 }
 
 /** Creates an <li> element containing text. */
-function createListElement(username, comment, timestamp) {
+function createListElement(username, comment, timestamp, imageUrl) {
   const liElement = document.createElement('li');
 
   const usernameElement = document.createElement('h5');
   const commentElement = document.createElement('p');
   const dateElement = document.createElement('h6');
+  const imgElement = document.createElement('img');
   usernameElement.innerText = username;
   commentElement.innerText = comment;
   dateElement.innerText = moment(timestamp).fromNow();
 
   liElement.appendChild(usernameElement);
   liElement.appendChild(commentElement);
+
+  if(imageUrl) {
+    imgElement.src = imageUrl;
+    liElement.appendChild(imgElement);
+  }
+
   liElement.appendChild(dateElement);
 
   return liElement;
 }
 
+/** Get blobstore upload url **/
+async function fetchBlobstoreUrl() {
+    const response = await fetch('/blobstore-upload-url');
+    const commentUploadUrl = await response.text();
```

Comment:
The response.text() needs an `await`?

Reply:
.text() returns a promise, thus i need to use await?

Reply:
Noted. Then it's fine.